Editor/Utility/CinemachineSceneTools.cs

```diff
@@ -0,0 +1,381 @@
+#if UNITY_2021_2_OR_NEWER
+using System;
+using System.Collections.Generic;
+using Cinemachine.Utility;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Experimental.Rendering;
+
+namespace Cinemachine.Editor
+{
+    /// <summary>
+    /// Static class that manages Cinemachine Tools. It knows which tool is active,
+    /// and ensures that no tools is active at the same time.
+    /// The tools and editors requiring tools register/unregister themselves here for control.
+    /// </summary>
+    static class CinemachineSceneToolUtility
+    {
+        /// <summary>
+        /// Checks whether tool is the currently active tool.
+        /// </summary>
+        /// <param name="tool">Tool to check.</param>
+        /// <returns>True, when the tool is the active tool. False, otherwise.</returns>
+        public static bool IsToolActive(Type tool)
+        {
+            return s_ActiveTool == tool;
+        }
+        static Type s_ActiveTool;
+
+        /// <summary>
+        /// Register your Type from the editor script's OnEnable function.
+        /// This way CinemachineTools will know which tools to display.
+        /// </summary>
+        /// <param name="tool">Tool to register</param>
+        public static void RegisterTool(Type tool)
+        {
+            s_RequiredTools.Add(tool);
+        }
+        
+        /// <summary>
+        /// Unregister your Type from the editor script's OnDisable function.
+        /// This way CinemachineTools will know which tools to display.
+        /// </summary>
+        /// <param name="tool">Tool to register</param>
+        public static void UnregisterTool(Type tool)
+        {
+            s_RequiredTools.Remove(tool);
+        }
+        static HashSet<Type> s_RequiredTools;
```

Comment:
What if 2 components add the same tool and then one of the components is changed?  It will incorrectly unregister the tool for both.  It would be more robust to have something like ref counting